### DISCUSSION TITLE
Qual a melhor linguagem para o desenvolvimento de um software em nuvem ?

### DISCUSSION THREAD

**Discussion by luismatheusbs16** [frontendbr/forum#1718] [Discussão]
**Hi coders!** Tenho uma ideia para criação de um software em nuvem, que faria a analise e gestão de dados de uma determinada empresa
a fonte de dados seria de múltiplos sites e arquivos onde o usurário normalmente armazenaria (excel, tableau,google analytics etc..),meu software 
faria a extração desse dados.
Ainda sou inciante em desenvolvimento, por isso gostaria ter uma direção certa da galera que já tem experiencia com desenvolvimento.
Para Back/Frontend qual a linguagem mais eficiente e estável eu deveria utilizar ? 
O Javascript (NodeJs, React Stack, e afins) é uma boa opção ?

**Comment by lorenacode** [comment]
Da uma olhada em wasm -> web assembly [Rust,Go]
https://developer.mozilla.org/en-US/docs/WebAssembly

**Comment by ninetails** [comment]
Para frontend ainda não tem como fugir muito do JS, mas aí é coisa de testar os 3 frameworks mais famosos e ver qual te facilita mais o trabalho ou qual deles vc entende mais

Já pra parte principal, falaria de JS por estar começando exceto se vc tiver alguma boa experiência em algo mais matemático, que se for o caso talvez o Python ajude por ter um ecossistema de libs que ajudem pra algumas fórmulas complexas, análise de dados etc...

disclaimer de sempre: Opiniões pessoais e não tome como verdades.

**Comment by gmtborges** [comment]
Tente usar a linguagem que se sente mais feliz/produtivo e tente aplicar as regras do 12factor
https://12factor.net/

**Comment by luismatheusbs16** [comment]
Assim, não importo muito com a questão de gosto @gustavomtborges , como o @ninetails disse o Js para front é quase inevitável , python seria interessante já que ele a linguagem do momento para analise de dados, e recursos não faltam, o meu real receio seria começar a desenvolver e chegar em uma etapa em que eu fale 'não consigo aplicar isso com essa linguagem' e meus avanços e se estagnarem

**Comment by ninetails** [comment]
@luismatheusbs16 também não necessariamente precisa ser em uma única linguagem de programação, mas aí começa a envolver arquitetura, microsserviços, custos, planejamento, investimento... algo que sai do controle de fazer sozinho e mais difícil ainda se estiver iniciando
